### PR TITLE
[Concept] CoordinatorV3

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: contracts/contracts/CoordinatorV3.sol [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    function getRites(uint32 ritualId) external view returns(Rite[] memory){
+        Rite[] memory rites = new Rite[](rituals[ritualId].rite.length);
+        for(uint32 i=0; i < rituals[ritualId].rite.length; i++){
+            rites[i] = rituals[ritualId].rite[i];
+        }
+        return rites;
+    }
+
+    function initiateRitual(address[] calldata nodes) external {
```
Comment (derekpierre) [inline contracts/contracts/CoordinatorV3.sol:92]:
```suggestion
    function initiateRitual(address[] calldata nodes) external returns (uint32) {
```

File: contracts/contracts/CoordinatorV3.sol [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        ritual.totalConfirmations++;
+
+        if (ritual.totalConfirmations == DKG_SIZE){
+            ritual.status = RitualStatus.COMPLETED;
+            emit EndRitual(ritualId, ritual.status);
+        }
+        emit AggregationPosted(ritualId, msg.sender, confirmedNodes);
+    }
+
+    function finalizeRitual(uint32 ritualId) {
```
Comment (derekpierre) [inline contracts/contracts/CoordinatorV3.sol:158]: 🔥

File: contracts/contracts/CoordinatorV3.sol [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+* @title CoordinatorV3
+* @notice Coordination layer for DKG-TDec
+*/
+contract CoordinatorV3 is Upgradeable {
+
+    // Ritual
+    event StartRitual(uint32 indexed ritualId, address[] nodes, address initiator);
+    event StartTranscriptRound(uint32 indexed ritualId);
+    event StartAggregationRound(uint32 indexed ritualId);
+    event EndRitual(uint32 indexed ritualId, RitualStatus status, address initiator);
```
Comment (theref) [inline contracts/contracts/CoordinatorV3.sol:17]: who is the `initiator` here?

**Comment by KPrasch** [comment]
subsumed by #59

**Pull request closed without merging**